table/block_based_table_builder.cc

```diff
@@ -78,8 +78,8 @@ FilterBlockBuilder* CreateFilterBlockBuilder(
       // as partition size.
       assert(table_opt.block_size_deviation <= 100);
       auto partition_size = static_cast<uint32_t>(
-          table_opt.metadata_block_size *
-          (100 - table_opt.block_size_deviation));
+          ((table_opt.metadata_block_size *
+          (100 - table_opt.block_size_deviation)) + 99) / 100);
```

Comment:
I don't quite understand the math. Can you give an example?

Reply:
metadata_block_size=1024
block_size_deviation=1
(1024 * (100 - 1)) / 100 = 1024 * 99 / 100 = 1013
The 99 is just to round up the number:
(1024 * (100 - 1) + 99) / 100 = (int)(1024 * 99 / 100.0 + 0.99) = 1014

Reply:
ok. but in what case it will end up with very large target size for filter partitions, without the round up?

Reply:
99 is trivial. The main change is "/ 100".

Reply:
ah